src/plugins/hetero/executable_network.cpp

```diff
@@ -58,16 +60,23 @@ using namespace InferenceEngine::HeteroConfigParams;
 template <typename T>
 using NodeMap = std::unordered_map<ngraph::Node*, T>;
 
-HeteroExecutableNetwork::HeteroExecutableNetwork(const InferenceEngine::CNNNetwork& network,
+HeteroExecutableNetwork::HeteroExecutableNetwork(const InferenceEngine::CNNNetwork& originalNetwork,
                                                  const Engine::Configs& config,
                                                  Engine* plugin)
     : InferenceEngine::ExecutableNetworkThreadSafeDefault(nullptr,
                                                           std::make_shared<InferenceEngine::ImmediateExecutor>()),
       _heteroPlugin{plugin},
-      _name{network.getName()},
+      _name{originalNetwork.getName()},
       _config{config} {
+
+    auto network = cloneNetwork(originalNetwork);
     auto function = network.getFunction();
     IE_ASSERT(function != nullptr);
+
+    ngraph::pass::Manager manager;
+    manager.register_pass<ngraph::pass::ConstantFolding>();
+    manager.run_passes(std::const_pointer_cast<ov::Model>(function));
+
     auto clonedFunction = ngraph::clone_function(*function);
```

Comment:
Could you use this cloned function?

Reply:
QueryNetwork for other plugin takes CNNNetwork as argument. So there are 2 ways how to handle it:
- clone CNNNetwork via cloneNetwork, perform constant folding on it, and send it to QueryNetwork
- clone ov::Model, perform constant folding on it, modify QueryNetwork to take ov::Model instead of CNNNetwork, and send modified ov::Model to it.

Reply:
You apply pass to ov::Model.
You can use folded ov::Model to create CNNNetwork

Reply:
done